ts/Stock/Indicators/RSI/RSIIndicator.ts

```diff
@@ -106,20 +107,28 @@ class RSIIndicator extends SMAIndicator {
             change: number,
             avgGain: number,
             avgLoss: number,
-            i: number;
+            i: number,
+            values: Array<number>;
 
         // RSI requires close value
-        if (
-            (xVal.length < period) || !isArray(yVal[0]) ||
-            yVal[0].length !== 4
-        ) {
+        if ((xVal.length < period)) {
             return;
         }
+        if (!isArray(yVal[0])) {
+            values = yVal as Array<number>;
+        } else if (yVal[0].length < 4) {
```

Comment:
Why `4`? What does it represent?

Reply:
As @raf18seb suggested earlier, it can prepare us to the cases, when you can define the RSI indicator for some other types of series like HLC, range etc.